src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -2942,7 +2942,7 @@ private void ValidateTransition(WriterState newState)
                             throw new ODataException(Strings.ODataWriterCore_InvalidTransitionFromResourceSet(this.State.ToString(), newState.ToString()));
                         }
 
-                        if (this.State == WriterState.DeletedResource && this.Version < ODataVersion.V401 && newState == WriterState.NestedResourceInfo)
+                        if (this.State == WriterState.DeletedResource && newState == WriterState.NestedResourceInfo)

```

Comment:
Won't this change be a breaking change?
Will this fix also affect requests, not only responses?

Reply:
it only affects responses not requests

Reply:
I believe this turns an error state into a success state for ODataVersion.V401, which is generally considered safe.